channeldb: specify freelist bbolt options by default

Crypt-iQ opened the pull request:
Fixes #3241.

These options could really be specified anywhere that bbolt is opened, so if that's of interest I can also do that.  But the only other large database I had on my machine was `wallet.db` which is the `btcwallet` repo.  Perhaps watchtowers might use a lot of space, or there could be some design parameters that change in the future where the other databases start using more space (macaroons.db etc).  Should I add these options to all the tests so they reflect the new behavior?

Updated bbolt dependency to v1.3.3 per https://github.com/etcd-io/bbolt/pull/153
The only downside is that if the db txn has to be rolled back in case of failure, the freelist must be built again temporarily which can be heavy with big databases.  I think this is an acceptable tradeoff since this shouldn't be happening too often.

cfromknecht commented on go.mod:
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/btcsuite/btcwallet v0.0.0-20190628225330-4a9774585e57
 	github.com/btcsuite/fastsha256 v0.0.0-20160815193821-637e65642941
-	github.com/coreos/bbolt v1.3.2
+	github.com/coreos/bbolt v1.3.3
```
small nit: swap the order of the commits so bbolt is updated first

Crypt-iQ commented:
@cfromknecht Addressed comment and added the options to watchtower client + server.  Can make a PR for btcwallet next.  For testing, I'm wondering if the options should also be specified in unit/itests?  If so, in order to reduce code duplication, I suppose I could add the options to `channeldb/options.go` similar to the way the cache sizes are setup?

cfromknecht commented:
@Crypt-iQ since the options are set inside the `Open` method, wouldn't these already be used in testing as well? Which cases do you think are still missing?

Crypt-iQ commented:
> @Crypt-iQ since the options are set inside the `Open` method, wouldn't these already be used in testing as well? Which cases do you think are still missing?

Hey you're right, I meant the other databases (sphinxreplay, macaroons) and some of the temporary test databases.  I didn't look at the results of my search very closely, mb

cfromknecht commented:
> Hey you're right, I meant the other databases (sphinxreplay, macaroons) and some of the temporary test databases. I didn't look at the results of my search very closely, mb

okay cool then i think we're good. sphinxreplay and macaroon dbs are less of a priority as they're much smaller, but not opposed to adding them as well

Crypt-iQ commented:
> > Hey you're right, I meant the other databases (sphinxreplay, macaroons) and some of the temporary test databases. I didn't look at the results of my search very closely, mb
> 
> okay cool then i think we're good. sphinxreplay and macaroon dbs are less of a priority as they're much smaller, but not opposed to adding them as well

The freelist on my 8MB sphinxreplay db has 5 free pages on it... so I don't think I'll include it here.  And my macaroons.db is tiny.

cfromknecht commented:
@Crypt-iQ decided we should push this off to 0.8, since it's a pretty big change for a point release and want to leave ample time for testing

Crypt-iQ commented:
@cfromknecht no problem, I'm looking into the InitialMmapSize option as well, so might be good to hold off until I can determine if it's worth adding

cfromknecht commented:
@Crypt-iQ sounds good, yeah we haven't done much exploration wrt bbolt options, would be interested to hear what you find!

Crypt-iQ commented:
Looked at `InitialMmapSize` and some of the related bbolt code.  Bbolt uses mmap to map the db file and sometimes it needs to be remapped via `munmap` and `mmap`.  In these cases, any write transactions that are referencing the old mmap need to copy these references to the heap.  So IMO it seems like we should just set a high `InitialMmapSize` and never have to remap.  Thoughts on size?

Crypt-iQ commented:
I'm kind of indifferent on the `InitialMmapSize` parameter.  I tested and couldn't, in any reasonable (non-test) circumstance, produce any huge memory allocations when resizing the mmap.  It is also not resized very often so it's not likely to occur that long-running reads (which there aren't that many of) block special write txn's.

Roasbeef commented:
> I'm kind of indifferent on the InitialMmapSize parameter.

I don't think we need to set this value in this PR, and can defer tuning it to a later PR. 

> But the only other large database I had on my machine was wallet.db which is the btcwallet repo.

Once this goes in, we can follow up with the change on the `btcwallet` side as well.
